async-aggregators/lib/hidden_in_ascii/hidden_in_ascii.py

```diff
@@ -0,0 +1,183 @@
+import random
+from pathlib import Path
+from typing import List, Tuple
+
+import numpy as np
+from PIL import Image, ImageDraw
+
+
+def prepare_input(img_path: Path) -> Tuple[Image.Image, List[int]]:
+    """
+    Prepare the input image by center cropping into a square
+
+    :param img_path: input image file path
+    :return: Tuple of cropped input Image and the coordinates of the cropped area
+    """
+    img = Image.open(img_path)
+
+    # Calculate the square size
+    width, height = img.size
+    size = min(width, height)
+
+    # Calculate the coordinates for cropping
+    left = (width - size) // 2
+    top = (height - size) // 2
+    right = (width + size) // 2
+    bottom = (height + size) // 2
+
+    # Crop the image to a square
+    return img.crop((left, top, right, bottom)), [left, top, right, bottom]
+
+
+def avg_greyscale(img: Image.Image) -> float:
+    """
+    Calculates average greyscale or luminance value of an input image
+
+    :param img: input image
+    :return: average luminance
+    """
+    np_img = np.array(img)
+    w, h = np_img.shape
+    return np.average(np_img.reshape(w * h))
+
+
+def img_to_ascii(img: Image.Image, dens: int) -> List[str]:
+    """
+    Given Image and dims (rows, cols) returns an m*n list of Images
+
+    :param img: input image file
+    :param dens: greyscale density level
+    :return: ascii text generated from input image
+    """
+    # The amount of text to white space in each of these character determines the greyscale value
+    greyscale_strs = [
+        "@%#*+=-:. ",  # 10 characters,
+        "@%W*hdwOLUzurt|1i}?+>,' ",  # 24 characters,
+        "@B$%8&WM#*oahkbdpqwmZO0QLCJUYXzcvunxrjft/\\|()1lIi!{}[]?-_+~<>;:,^`'. ",  # 70 characters,
+    ]
+
+    # Convert to greyscale (L stands for luminance)
+    gs_img = img.convert("L")
+    gs_img_w, gs_img_h = gs_img.size[0], gs_img.size[1]
+    cols = int(gs_img_w // 10)  # 1/10th of the cropped square
+    scale = 0.39  # Aspect ratio of monospace characters
+
+    w = gs_img_w / cols  # tile width
+    h = w / scale  # tile height
+    rows = int(gs_img_h / h)  # Number of rows of text
+
+    # Generate ascii art
+    ascii_ = []
```

Comment:
When initializing an empty list like this, it's good to typehint the values that will be added to it later.
```suggestion
    ascii_: list[str] = []
```